src/store/participantsStore.spec.js

```diff
@@ -572,27 +572,109 @@ describe('participantsStore', () => {
 			})
 		})
 
-		test('joins call', async () => {
-			// Assert
+		afterEach(() => {
+			jest.clearAllTimers()
+		})
+
+		const assertInitialCallState = () => {
 			expect(joinCall).toHaveBeenCalledWith(TOKEN, flags, false, false)
+			EventBus.emit('signaling-join-call', [TOKEN, actualFlags])
 			expect(store.getters.isInCall(TOKEN)).toBe(true)
 			expect(store.getters.isConnecting(TOKEN)).toBe(true)
-			expect(store.getters.participantsList(TOKEN)).toStrictEqual([
-				{
-					attendeeId: 1,
-					sessionId: 'session-id-1',
-					inCall: actualFlags,
-					participantType: PARTICIPANT.TYPE.USER,
-				},
-			])
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+			expect(store.getters.participantsList(TOKEN)).toStrictEqual([{
+				attendeeId: 1,
+				sessionId: 'session-id-1',
+				inCall: actualFlags,
+				participantType: PARTICIPANT.TYPE.USER,
+			}])
+		}
 
-			// Finished connecting to the call
-			EventBus.emit('signaling-users-in-room')
+		const finishConnectingWithUsers = (participants) => {
+			EventBus.emit('signaling-users-in-room', [participants])
+			expect(store.getters.isInCall(TOKEN)).toBe(true)
+			expect(store.getters.isConnecting(TOKEN)).toBe(false)
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+		}
+
+		const emitFailureAndAssert = (reason) => {
+			EventBus.emit('signaling-join-call-failed', [TOKEN, reason])
+			expect(store.getters.isInCall(TOKEN)).toBe(false)
+			expect(store.getters.isConnecting(TOKEN)).toBe(false)
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+			expect(store.getters.connectionFailed(TOKEN)).toBe(reason)
+		}
 
+		const assertParticipantsListReceivedBeforeJoiningCall = (event, payload) => {
+			EventBus.emit(event, [payload])
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(true)
+			expect(store.getters.isConnecting(TOKEN)).toBe(true)
+			expect(store.getters.isInCall(TOKEN)).toBe(false)
+			// join call
+			EventBus.emit('signaling-join-call', [TOKEN, actualFlags])
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+			expect(store.getters.isConnecting(TOKEN)).toBe(false)
+			expect(store.getters.isInCall(TOKEN)).toBe(true)
+		}
+
+		test('joins call with internal signaling', async () => {
+			assertInitialCallState()
+			finishConnectingWithUsers([{
+				attendeeId: 1,
+				sessionId: 'session-id-1',
+				inCall: actualFlags,
+				participantType: PARTICIPANT.TYPE.USER,
+			}])
+		})
+
+		test('joins call with external signaling', async () => {
+			assertInitialCallState()
+			EventBus.emit('signaling-users-changed', [[{
+				attendeeId: 1,
+				nextcloudSessionId: 'session-id-1',
+				inCall: actualFlags,
+				participantType: PARTICIPANT.TYPE.USER,
+			}]])
+			expect(store.getters.isInCall(TOKEN)).toBe(true)
+			expect(store.getters.isConnecting(TOKEN)).toBe(false)
+			expect(store.getters.isJoiningCall(TOKEN)).toBe(false)
+		})
+
+		test('joins call but it fails', async () => {
+			assertInitialCallState()
+			emitFailureAndAssert('error reason')
+		})
```

Comment:
Error handling is here, so you don't receive flags
```suggestion
		})
		
		test('trying to join call but it fails', async () => {
			emitFailureAndAssert('error reason')
		})
```